exchanges/poloniex/poloniex.go

```diff
@@ -16,920 +16,1116 @@ import (
-	if amount != 0 {
-		values.Set("amount", strconv.FormatFloat(amount, 'f', -1, 64))
+	var resp *PlaceOrderResponse
+	if err := e.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, authResourceIntensiveEPL, http.MethodPost, "/orders", nil, arg, &resp); err != nil {
+		return nil, err
 	}
-
-	err := e.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, http.MethodPost,
-		poloniexOrderMove,
-		values,
-		&result)
-	if err != nil {
-		return result, err
+	if resp == nil {
+		return nil, common.ErrNoResponse
+	} else if resp.Code != 0 && resp.Code != 200 {
+		return resp, fmt.Errorf("%w: code: %d message: %s", common.ErrNoResponse, resp.Code, resp.Message)
 	}
+	return resp, nil
+}
 
-	if result.Success != 1 {
-		return result, errors.New(result.Error)
+// PlaceBatchOrders places a batch of orders
+func (e *Exchange) PlaceBatchOrders(ctx context.Context, args []PlaceOrderRequest) ([]*PlaceBatchOrderItem, error) {
+	if len(args) == 0 {
+		return nil, common.ErrNilPointer
 	}
-
-	return result, nil
+	for x := range args {
+		if args[x].Symbol.IsEmpty() {
+			return nil, currency.ErrCurrencyPairEmpty
+		}
+		if args[x].Side == "" {
+			return nil, order.ErrSideIsInvalid
+		}
+		if args[x].Amount <= 0 {
+			return nil, limits.ErrAmountBelowMin
+		}
+	}
+	var resp []*PlaceBatchOrderItem
+	return resp, e.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, authResourceIntensiveEPL, http.MethodPost, "/orders/batch", nil, args, &resp)
 }
 
-// Withdraw withdraws a currency to a specific delegated address.
-// For currencies where there are multiple networks to choose from (like USDT or BTC),
-// you can specify the chain by setting the "currency" parameter to be a multiChain currency
-// name, like USDTTRON, USDTETH, or BTCTRON
-func (e *Exchange) Withdraw(ctx context.Context, ccy, address string, amount float64) (*Withdraw, error) {
-	result := &Withdraw{}
-	values := url.Values{}
-
-	values.Set("currency", strings.ToUpper(ccy))
-	values.Set("amount", strconv.FormatFloat(amount, 'f', -1, 64))
-	values.Set("address", address)
-
-	err := e.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, http.MethodPost, poloniexWithdraw, values, &result)
-	if err != nil {
+// CancelReplaceOrder cancels an existing active order, new or partially filled, and places a new order
+func (e *Exchange) CancelReplaceOrder(ctx context.Context, arg *CancelReplaceOrderRequest) (*CancelReplaceOrderResponse, error) {
+	if arg.OrderID == "" {
+		return nil, order.ErrOrderIDNotSet
+	}
+	var resp *CancelReplaceOrderResponse
+	if err := e.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, authNonResourceIntensiveEPL, http.MethodPut, "/orders/"+arg.OrderID, nil, arg, &resp); err != nil {
 		return nil, err
 	}
-
-	if result.Error != "" {
-		return nil, errors.New(result.Error)
+	if resp == nil {
+		return nil, common.ErrNoResponse
+	} else if resp.Code != 0 && resp.Code != 200 {
+		return resp, fmt.Errorf("%w: code: %d message: %s", common.ErrNoResponse, resp.Code, resp.Message)
 	}
```

Comment:
🚧 Change request: (Reposted)
I don't think that common.ErrNoResponse is correct here.
Can we not tell the user anything useful about which order just failed?
ClientOrderID and index would be a good idea.